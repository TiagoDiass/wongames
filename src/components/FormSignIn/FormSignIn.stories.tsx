import { ComponentMeta, ComponentStory } from '@storybook/react';
import FormSignIn from './FormSignIn';

export default {
  title: 'Form/FormSignIn',
  component: FormSignIn
} as ComponentMeta<typeof FormSignIn>;

export const Basic: ComponentStory<typeof FormSignIn> = (args) => (
  <div style={{ width: 300, margin: 'auto' }}>
    <FormSignIn {...args} />
  </div>
);
