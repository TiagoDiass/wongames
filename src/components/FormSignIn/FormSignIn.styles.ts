import styled, { css } from 'styled-components';

export const ForgotPasswordLink = styled.a`
  ${({ theme }) => css`
    display: block;
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.black};
    text-decoration: none;
    text-align: right;

    &:hover {
      text-decoration: underline;
    }
  `}
`;
