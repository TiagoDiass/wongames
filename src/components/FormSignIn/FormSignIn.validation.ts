import { useMemo } from 'react';
import * as yup from 'yup';
import { yupResolver } from '@hookform/resolvers/yup';

const validationSchema = yup.object().shape({
  email: yup.string().required('Required field').email('Invalid email'),
  password: yup
    .string()
    .required('Required field')
    .min(8, 'Password must have at least 8 characters')
});

export const useValidationSchema = () => useMemo(() => yupResolver(validationSchema), []);
