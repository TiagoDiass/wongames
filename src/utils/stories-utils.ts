type Backgrounds = 'Dark' | 'Light' | 'Light Blue';

export const setStoryDefaultBackground = (background: Backgrounds) => ({
  backgrounds: {
    default: background
  }
});

type Layouts = 'fullscreen' | 'centered' | 'padded';

export const setStoryLayout = (storyLayout: Layouts) => ({
  layout: storyLayout
});
