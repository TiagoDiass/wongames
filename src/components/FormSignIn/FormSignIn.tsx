import { Button, TextField } from 'components';
import Link from 'next/link';
import { Email as EmailIcon, Lock as LockIcon } from 'styled-icons/material-outlined';

import * as S from './FormSignIn.styles';

/**
 * Sign In Form
 */
export default function FormSignIn() {
  return (
    <S.Wrapper>
      <form>
        <TextField name='email' placeholder='Email' type='email' icon={<EmailIcon />} />
        <TextField name='password' placeholder='Password' type='password' icon={<LockIcon />} />

        <S.ForgotPasswordLink href='#'>Forgot your password?</S.ForgotPasswordLink>

        <Button size='large' fullWidth>
          Sign in now
        </Button>

        <S.FormLinkWrapper>
          {"Don't have an account?"}
          <Link href='/sign-up'>
            <a>Sign up</a>
          </Link>
        </S.FormLinkWrapper>
      </form>
    </S.Wrapper>
  );
}
