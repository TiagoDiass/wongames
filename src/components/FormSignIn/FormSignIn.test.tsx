import FormSignIn from './FormSignIn';
import { renderWithTheme } from 'utils/test-utils';
import { screen } from '@testing-library/react';

describe('Component: FormSignIn', () => {
  it('should render correctly (fields and secondary links)', () => {
    const { container } = renderWithTheme(<FormSignIn />);

    expect(screen.getByPlaceholderText('Email')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Password')).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Sign in now' })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: 'Forgot your password?' })).toBeInTheDocument();
    expect(screen.getByText("Don't have an account?")).toBeInTheDocument();
    expect(screen.getByRole('link', { name: 'Sign up' })).toBeInTheDocument();

    expect(container.parentElement).toMatchSnapshot();
  });
});
