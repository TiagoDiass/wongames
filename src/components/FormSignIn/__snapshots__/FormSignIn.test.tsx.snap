// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: FormSignIn should render correctly (fields and secondary links) 1`] = `
.c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: none;
  border-radius: 0.4rem;
  outline: none;
  padding: 0.8rem;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
  width: 100%;
}

.c8:disabled {
  opacity: 0.7;
  cursor: not-allowed;
}

.c8:disabled:hover {
  -webkit-filter: none;
  filter: none;
}

.c8:focus {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c8:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.c4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #EAEAEA;
  border-radius: 0.2rem;
  padding: 0 0.8rem;
  border: 0.2rem solid;
  border-color: #EAEAEA;
  -webkit-transition: box-shadow 0.3s ease-in-out;
  transition: box-shadow 0.3s ease-in-out;
}

.c2:focus-within {
  border: 1px solid #F231A5;
}

.c5 {
  color: #030517;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  padding: 0.8rem;
  background: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

.c5:-webkit-autofill,
.c5:-webkit-autofill:hover,
.c5:-webkit-autofill:focus,
.c5:-webkit-autofill:active {
  -webkit-box-shadow: 0 0 0 30px #EAEAEA inset !important;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 2.2rem;
  color: #8F8F8F;
}

.c3 > svg {
  width: 100%;
}

.c0 .c1 {
  margin: 0.8rem 0;
}

.c0 .c7 {
  margin: 3.2rem auto 1.6rem;
}

.c9 {
  font-size: 1.4rem;
  color: #030517;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 4px;
  column-gap: 4px;
}

.c9 a {
  color: #3CD3C1;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 0.1rem solid #3CD3C1;
  -webkit-transition: filter,0.1s ease-in-out;
  transition: filter,0.1s ease-in-out;
}

.c9 a:hover {
  -webkit-filter: brightness(0.85);
  filter: brightness(0.85);
}

.c6 {
  display: block;
  font-size: 1.4rem;
  color: #030517;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: right;
}

.c6:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<body>
  <div>
    <div
      class="c0"
    >
      <form>
        <div
          class="c1 "
        >
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <svg
                aria-hidden="true"
                class="c4"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6zm-2 0l-8 5-8-5h16zm0 12H4V8l8 5 8-5v10z"
                />
              </svg>
            </div>
            <input
              aria-disabled="false"
              aria-invalid="false"
              class="c5"
              name="email"
              placeholder="Email"
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          class="c1 "
        >
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <svg
                aria-hidden="true"
                class="c4"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                  />
                  <path
                    d="M0 0h24v24H0V0z"
                    opacity="0.87"
                  />
                </g>
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
                />
              </svg>
            </div>
            <input
              aria-disabled="false"
              aria-invalid="false"
              class="c5"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
        </div>
        <a
          class="c6"
          href="#"
        >
          Forgot your password?
        </a>
        <button
          class="c7 c8"
        >
          <span>
            Sign in now
          </span>
        </button>
        <div
          class="c9"
        >
          Don't have an account?
          <a
            href="/sign-up"
          >
            Sign up
          </a>
        </div>
      </form>
    </div>
  </div>
</body>
`;
